### PR TITLE
WT-5037 Move developer documentation from github wiki to source.wiredtiger.com

### PR DESCRIPTION
**Pull request by tammybailey**
I merged four of the pages in the "Application Developer Resources" section of the [WiredTiger wiki](https://github.com/wiredtiger/wiredtiger/wiki) to the "Developer Documentation" section of source.wiredtiger.com. I did not add the three pages of the wiki related to `t2` as this is an internal MongoDB tool so it's not useful information for the general public.

The plan is to delete the reproduced wiki pages after this PR is approved.

### CODE REVIEW

**Comment by wtbuild** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by lukech** [comment]
test this please

File: src/docs/devdoc-index.dox [hunk 1]
```diff
@@ -1,22 +1,45 @@
...
 for the first time. It is kept in a global list and is shared across the
 sessions.  When a dhandle is not needed anymore and has been idle for a while,
 it is closed and destroyed, releasing all the resources associated with it.
 
-For more information on the lifecycle of a dhandle, see:
+@subpage devdoc-statistics
+
+WiredTiger can generate statistics that are useful for providing information
+necessary when performance tuning your WiredTiger application. Here we focus on
+on analyzing and reviewing the data generated by the statistics logging
```
Comment (quchenhao) [inline src/docs/devdoc-index.dox]: redundant `on`

File: src/docs/devdoc-statistics.dox [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+(http://source.wiredtiger.com/2.0.1/statistics.html) page in the WiredTiger
+documentation. This content focuses on analyzing and reviewing the data
+generated by the statistics logging functionality.
+
+# Enabling statistics
+
+By default statistics are disabled in WiredTiger - this is because tracking
+statistics has a performance overhead. WiredTiger has two modes for tracking
+statistics `fast` and `all`. Fast statistics are those that don't incur a
+significant performance overhead. The affect on performance of enabling
```
Comment (quchenhao) [inline src/docs/devdoc-statistics.dox]: effect

Comment (tammybailey) [reply src/docs/devdoc-statistics.dox]: Fixed.

File: src/docs/devdoc-index.dox (hunk 1 above)
Comment (tammybailey) [reply src/docs/devdoc-index.dox]: Fixed.

**Pull request merged**